server/netty-server/cats/src/test/scala/sttp/tapir/server/netty/cats/NettyCatsRequestTimeoutTest.scala

```diff
@@ -0,0 +1,95 @@
+package sttp.tapir.server.netty.cats
+
+import cats.effect.{IO, Resource}
+import cats.effect.std.Dispatcher
+import cats.effect.unsafe.implicits.global
+import io.netty.channel.EventLoopGroup
+import org.scalatest.matchers.should.Matchers._
+import sttp.capabilities.WebSockets
+import sttp.capabilities.fs2.Fs2Streams
+import sttp.client3._
+import sttp.model.HeaderNames
+import sttp.tapir._
+import sttp.tapir.server.netty.NettyConfig
+import sttp.tapir.tests.Test
+
+import scala.concurrent.duration.DurationInt
+
+class NettyCatsRequestTimeoutTest(
+    dispatcher: Dispatcher[IO],
+    eventLoopGroup: EventLoopGroup,
+    backend: SttpBackend[IO, Fs2Streams[IO] with WebSockets]
+) {
+  def tests(): List[Test] = List(
+    Test("chunked transmission lasts longer than given timeout") {
+      val givenRequestTimeout = 2.seconds
+      val howManyChunks: Int = 2
+      val chunkSize = 100
+      val millisBeforeSendingSecondChunk = 1000L
+
+      val e =
+        endpoint.post
+          .in(header[Long](HeaderNames.ContentLength))
+          .in(streamTextBody(Fs2Streams[IO])(CodecFormat.TextPlain()))
+          .out(header[Long](HeaderNames.ContentLength))
+          .out(streamTextBody(Fs2Streams[IO])(CodecFormat.TextPlain()))
+          .serverLogicSuccess[IO] { case (length, stream) =>
+            IO((length, stream))
+          }
+
+      val config =
+        NettyConfig.default
+          .eventLoopGroup(eventLoopGroup)
+          .randomPort
+          .withDontShutdownEventLoopGroupOnClose
+          .noGracefulShutdown
+          .requestTimeout(givenRequestTimeout)
+
+      val bind = NettyCatsServer(dispatcher, config).addEndpoint(e).start()
+
+      def iterator(howManyChunks: Int, chunkSize: Int): Iterator[Byte] = new Iterator[Iterator[Byte]] {
+        private var chunksToGo: Int = howManyChunks
+
+        def hasNext: Boolean = {
+          if (chunksToGo == 1)
+            Thread.sleep(millisBeforeSendingSecondChunk)
```

Comment:
when using fs2 you shouldn't use Thread.sleep etc., instead you should create the stream using fs2 combinators (without going through iterators etc.)